### PR TITLE
ddl: Add lower limit of length of index column type

### PR DESCRIPTION
**Pull request by jackysp**
Fix issue #3833 . @zimulala @hicqu PTAL

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -49,6 +49,10 @@ func buildIndexColumns(columns []*model.ColumnInfo, idxColNames []*ast.IndexColN
 			return nil, errKeyColumnDoesNotExits.Gen("column does not exist: %s", ic.Column.Name)
 		}
 
+		if col.Flen == 0 {
```
Comment (shenli) [inline ddl/index.go]: We should check this in plan/validator.go.

Comment (jackysp) [inline ddl/index.go]: How to check table column type length in `create index` grammar?

Comment (jackysp) [inline ddl/index.go]: It is not a grammar check here. #3838 and #3835 will be fixed in plan/validator.go

Comment (shenli) [inline ddl/index.go]: You can check the Flen in this function:https://github.com/pingcap/tidb/blob/master/plan/validator.go#L259

Refer to [this](https://github.com/pingcap/tidb/blob/master/plan/validator.go#L217) as an example.

Comment (jackysp) [inline ddl/index.go]: A statement as `create index idx on t (i)` will not pass table column type, it is not the same case of `create table` statement.

**Comment by shenli** [comment]
LGTM

**Comment by shenli** [comment]
@tiancaiamao @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**